carapace-server/src/main/java/org/carapaceproxy/SimpleHTTPResponse.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed to Diennea S.r.l. under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Diennea S.r.l. licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ */
+package org.carapaceproxy;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import org.carapaceproxy.server.mapper.CustomHeader;
+
+/**
+ * Static response to sent to clients.
+ *
+ * @author paolo.venturi
+ */
+public class SimpleHTTPResponse {
+
+    private final int errorcode;
+    private final String resource;
+    private final List<CustomHeader> customHeaders;
+
+    public SimpleHTTPResponse(int errorcode, String resource, List<CustomHeader> customHeaders) {
+        this.errorcode = errorcode;
+        this.resource = resource;
+        this.customHeaders = Collections.unmodifiableList(customHeaders == null ? new ArrayList() : customHeaders);
```

Comment:
nit:
`Collections.unmodifiableList(customHeaders == null ? new ArrayList()`

why not this form ? no memory allocations
`Collections.emptyList()`